examples/cc3200/xively_demo/main.c

```diff
@@ -0,0 +1,1044 @@
+                    XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    }
+    else
+    {
+        xi_publish( gXivelyContextHandle, gButtonSW3TopicName, "1",
+                    XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    }
+
+    Button_IF_EnableInterrupt( SW3 );
+}
+
+/**
+ * @brief Function that handles events related to the LEDs' topics messages and
+ * subscription status changes.
+ *
+ * @note For more details please refer to the xively.h - xi_subscribe function.
+ *
+ * @param in_context_handle - xively library context required for calling API functions
+ * @param call_type - describes if the handler was called carrying the subscription
+ * operation result or the incoming message
+ * @oaram params - structure with data
+ * @param state - state of the operation
+ * @oaram user_data - pointer attached to the topic
+ */
+void onLedTopic( xi_context_handle_t in_context_handle,
+                 xi_sub_call_type_t call_type,
+                 const xi_sub_call_params_t* const params,
+                 xi_state_t state,
+                 void* user_data )
+{
+    switch ( call_type )
+    {
+        case XI_SUB_CALL_SUBACK:
+            if ( params->suback.suback_status == XI_MQTT_SUBACK_FAILED )
+            {
+                UART_PRINT( "topic:%s. Subscription failed.\n", params->suback.topic );
+            }
+            else
+            {
+                UART_PRINT( "topic:%s. Subscription granted %d.\n", params->suback.topic,
+                            ( int )params->suback.suback_status );
+            }
+            return;
+        case XI_SUB_CALL_MESSAGE:
+        {
+            if ( params->message.temporary_payload_data_length == 1 )
+            {
+                ledNames ledName = *( ( ledNames* )user_data );
+
+                switch ( params->message.temporary_payload_data[0] )
+                {
+                    case 48:
+                        UART_PRINT( "topic:%s off %d\r\n", params->suback.topic,
+                                    ( int )ledName );
+                        GPIO_IF_LedOff( ledName );
+                        break;
+                    case 49:
+                        UART_PRINT( "topic:%s on %d\r\n", params->suback.topic,
+                                    ( int )ledName );
+                        GPIO_IF_LedOn( ledName );
+                        break;
+                    default:
+                        UART_PRINT( "unexpected value on topic %s \r\n",
+                                    params->message.topic );
+                        break;
+                }
+            }
+            else
+            {
+                UART_PRINT( "unexpected data length on topic %s \r\n",
+                            params->message.topic );
+            }
+        }
+            return;
+        default:
+            return;
+    }
+}
+
+/**
+ * @brief Implementation of recurring time task that sends updates on temperature sensor

```

Comment:
Recurring time task that activates temperature sensor and publishes the result on the temperature topic for the device.